### DISCUSSION TITLE
After setting `foldexpr`, the code is folded every time entering a buffer.

### DISCUSSION THREAD

**Discussion by ybisland** [nvim-treesitter/nvim-treesitter#2903] [Q&A] [answered]
After I put the two following settings into my config files, the code gets collapsed every time I enter the buffer.

**minimal config:**
```lua
vim.cmd [[ set foldmethod=expr ]]
vim.cmd [[ set foldexpr=nvim_treesitter#foldexpr() ]]
```

**code gets collapsed when entering buffers:**
![OB)W@935GI7200~34I030HU](https://user-images.githubusercontent.com/37294860/166710782-93082006-5e8f-4ab8-b691-832b26d7af34.png)

It seems that every time I enter a buffer, `nvim_treesitter#foldexpr()` is automatically performed. 

**expect:**

I hope code is not collapsed when I enter a buffer. But if I don't set `foldexpr`, I cannot fold code in neovim. Is there a way to avoid this?

**Comment by fsouza** [comment, marked as answer]
You can probably set `foldlevelstart` to some high value like 99, and that should work.

**Reply by ybisland** [reply]
thanks, it works!
